xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp

```diff
@@ -2046,13 +2044,8 @@ void COutput::InitCycle()
 
   EINTERLACEMETHOD method = m_config.processInfo->GetVideoSettings().m_InterlaceMethod;
   bool interlaced = m_currentPicture.DVDPic.iFlags & DVP_FLAG_INTERLACED;
-  // Remember whether any interlaced frames were encountered already.
-  // If this is the case, the deinterlace method will never automatically be switched to NONE again in
-  // order to not change deint methods every few frames in PAFF streams.
-  m_seenInterlaced = m_seenInterlaced || interlaced;
 
-  if (!(flags & DVD_CODEC_CTRL_NO_POSTPROC) &&
-      m_seenInterlaced &&
+  if (!(flags & DVD_CODEC_CTRL_NO_POSTPROC) && interlaced && interlaced &&
```

Comment:
Shouldn't this be just
if (!(flags & DVD_CODEC_CTRL_NO_POSTPROC) && interlaced &&

Rather than testing the interlaced flag twice